### PR TITLE
add resolver api service and ci cd

### PR DESCRIPTION
**Pull request by jyothishnt**
Review app endpoint 
http://add-api-endpoint.review.ensembl.org/api/id/ENSG00000127720

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+.deploy:
+  stage: deploy
+  image: dockerhub.ebi.ac.uk/ensembl-web/deploy-tools:latest
+  before_script:
+    - kubectl config use-context ${AGENT}
+    - kubectl config set-context --current --namespace=${NS}  
+  script:
+    - git clone https://gitlab.ebi.ac.uk/ensembl-web/ensembl-k8s-manifests.git
+    - git -C ensembl-k8s-manifests/ checkout master
```
Comment (kamaldodiya) [inline .gitlab-ci.yml:39]: This should be k8s123-migration branch not master

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+COPY requirements.txt requirements.txt
+# Install dependencies
+RUN pip install  -r requirements.txt
+
+# Expose Ports
+ENV PORT 8080
+EXPOSE 8080
+
+# Run uvicorn server
+CMD ["fastapi", "run", "main.py", "--port", "8080"]
```
Comment (kamaldodiya) [inline Dockerfile:21]: Use uvicorn server not fastapi. I think fastapi is good for development not for production use cases

**Review by kamaldodiya — CHANGES_REQUESTED** [review]

File: app/api/models/resolver.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from typing import Optional, Literal, List
+from pydantic import BaseModel, validator, Field
+
+
+class SearchPayload(BaseModel):
+  stable_id: str = Field(default = None, title = "Stable ID of a gene")
+  type: Optional[Literal['gene']] = Field(default = None, title = "Type of stable id, e.g. gene" )
+  genome_ids: List = []
+  per_page: int = 1
+  # gca: Optional[str] = Field (default = None, title = "GCA accession id for the genome")
```
Comment (Mehrnaz-Charkhchi) [inline app/api/models/resolver.py:10]: Please remove comments.

**Review by kamaldodiya — APPROVED** [review]

File: app/api/models/resolver.py (hunk 3 above)
Comment (jyothishnt) [reply app/api/models/resolver.py:10]: This is something I kept for future. Should I remove it?

**Review by Mehrnaz-Charkhchi — APPROVED** [review]

File: app/api/error_response.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+def response_error_handler(result):
+    if result["status"] == 501:
+        return http_501_not_implemented()
+    if result["status"] == 500:
+        return http_500_internal_server_error()
+    if result["status"] == 406:
+        return http_406_not_acceptable()
+    if result["status"] == 400:
+        return http_400_bad_request()
+    if result["status"] == 501:
```
Comment (veidenberg) [inline app/api/error_response.py:22]: Duplicate

File: app/api/models/resolver.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from typing import Optional, Literal, List
+from pydantic import BaseModel, validator, Field
+
+
+class SearchPayload(BaseModel):
+  stable_id: str = Field(default = None, title = "Stable ID of a gene")
+  type: Optional[Literal['gene']] = Field(default = None, title = "Type of stable id, e.g. gene" )
+  genome_ids: List = []
```
Comment (veidenberg) [inline app/api/models/resolver.py:8]: You can use builtin `list` type (like you do in `config.py` line 37)

**Pull request merged**